src/status_im2/contexts/add_new_contact/events.cljs

```diff
@@ -0,0 +1,96 @@
+(ns status-im2.contexts.add-new-contact.events
+  (:require [utils.re-frame :as rf]
+            [status-im.utils.types :as types]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.ens :as ens]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.native-module.core :as status]
+            [status-im.utils.db :as utils.db]
+            [status-im.utils.utils :as utils]))
+
+(re-frame/reg-fx
+ :contacts/resolve-public-key-from-ens-name
+ (fn [{:keys [chain-id ens-name on-success on-error]}]
+   (ens/pubkey chain-id ens-name on-success on-error)))
+
+(re-frame/reg-fx
+ :contacts/decompress-public-key
+ (fn [{:keys [public-key on-success on-error]}]
+   (status/compressed-key->public-key
+    public-key
+    (fn [resp]
+      (let [{:keys [error]} (types/json->clj resp)]
+        (if error
+          (on-error error)
+          (on-success
+           (str "0x" (subs resp 5)))))))))
+
+(rf/defn set-new-identity
+  {:events [:contacts/set-new-identity]}
+  [{:keys [db]} input]
+  (let [input           (utils/safe-trim input)
+        public-key?     (utils.db/valid-public-key? input)
+        compressed-key? (utils.db/valid-compressed-key? input)
+        ens-name        (if compressed-key? nil (stateofus/ens-name-parse input))
+        on-success      (fn [pubkey]
+                          (rf/dispatch
+                           [:contacts/set-new-identity-success
+                            input ens-name pubkey]))
+        on-error        (fn [err]
+                          (rf/dispatch
+                           [:contacts/set-new-identity-error err input]))]
+    (cond
+      (empty? input) {:db (dissoc db :contacts/new-identity)}
+      public-key?    {:db (assoc db
+                                 :contacts/new-identity
+                                 {:input      input
+                                  :public-key input
+                                  :ens-name   nil
+                                  :state      :error
+                                  :error      :uncompressed-key})}
+      :else
+      (conj {:db (assoc db
```

Comment:
better to use cond-> and assoc, easier to read `(conj {} [])` looks strange

Reply:
hmm... I kinda liked this: just one conj with no threading. And I like seeing key-values in vectors since it kind of bundles them together. But there's a lot more `assoc`'ing in the codebase, so I agree with updating this. Thanks!

Reply:
I think this is a nitpick, both options work as you like, I'm more concerned about how `(conj {} [])` works, why you conj a vector, it's kind unexpected that this works

Reply:
so from my clojure knowlage base `(conj {} [])` something alien , no?

Reply:
`(conj {} {})` seems more readable

Reply:
> so from my clojure knowlage base (conj {} []) something alien , no?

It's the first time I'm seeing this as well. Got me thinking

Reply:
```clojure
cljs.user=> (conj {:a 1} [:b 2 :c 3])
{:a 1, :b 2}
cljs.user=> (conj {:a 1} {:b 2 :c 3})
{:a 1, :c 3, :b 2}
```

Reply:
key-values are added as separate vectors [ClojureDocs](https://clojuredocs.org/clojure.core/conj#example-547dbee4e4b03d20a10242bb)
```clojure
cljs.user> (conj {:a 1} [:b 2] [:c 3])
{:a 1, :b 2, :c 3}
```
I used clojure mostly for data transformation projects, I'm not sure where/why but this seems familiar to me. I'm comfortable with assoc too :)

Reply:
Just so you can fully understand my preference, it's not that I just prefer `conj` over `assoc` in this case.

It's that I have to think about the threading macro too: In my head, I think "here's a threading macro, it's the `cond` type, which means I'm going to be looking for conditions on what operations to apply to the first argument, which is a map. Then after the conditions there are operations (looks like they are all `assoc`'s), and then I have to think about where the first argument will be inserted in that assoc operation (sometimes it doesn't work based on the operation, in which case I'd use `as->`).

vs. "ok, this is `conj`'ing stuff onto a map. Looks like they're vectors, which are key-values in the map. Now there's some conditional if/whatever for each of those vectors."

For me, once I see that it's a `(conj map vector1 vector2 ...)` structure, it's an easy thing for me to reason about (I know at a high-level we're only going to be adding key-values to a map, nothing else). The threading macro approach makes me think about operations (assoc's) all the way down the threading macro.